src/resource-graph/azext_resourcegraph/custom.py

```diff
@@ -14,90 +14,95 @@
-                result_format=ResultFormat.object_array
-            )
-
-            request = QueryRequest(query=full_query, subscriptions=subs_list, options=request_options)
-            response = client.resources(request)  # type: QueryResponse
-            if response.result_truncated == ResultTruncated.true:
-                result_truncated = True
 
+        if first is not None:
+            first = min(first, __ROWS_PER_PAGE)
+        elif skip_token is None:
+            first = __FIRST
+
+        if skip is None and skip_token is None:
+            skip = __SKIP
+
+        request_options = QueryRequestOptions(
+            top=first,
+            skip=skip,
+            skip_token=skip_token,
+            result_format=ResultFormat.object_array,
+            allow_partial_scopes=allow_partial_scopes
+        )
+
+        request = QueryRequest(
+            query=graph_query,
+            subscriptions=subs_list,
+            management_groups=mgs_list,
+            options=request_options)
+        response = client.resources(request)  # type: QueryResponse
+        if response.result_truncated == ResultTruncated.true:
+            result_truncated = True
+
+        results.extend(response.data)
+
+        if hasattr(response, 'skip_token'):
             skip_token = response.skip_token
-            results.extend(response.data)
-
-            if len(results) >= first or skip_token is None:
-                break
 
         if result_truncated and len(results) < first:
             __logger.warning("Unable to paginate the results of the query. "
                              "Some resources may be missing from the results. "
                              "To rewrite the query and enable paging, "
                              "see the docs for an example: https://aka.ms/arg-results-truncated")
 
-    except ErrorResponseException as ex:
-        raise CLIError(json.dumps(_to_dict(ex.error), indent=4))
-
-    return results
-
+    except HttpResponseError as ex:
+        raise CLIError(json.dumps(_to_dict(ex.model.error), indent=4))
 
-def create_shared_query(client, resource_group_name,
-                        resource_name, description,
-                        graph_query, location='global', tags=None):
-    from azext_resourcegraph.vendored_sdks.resourcegraph.models import GraphQueryResource
-    graph_shared_query = GraphQueryResource(description=description,
-                                            query=graph_query,
-                                            tags=tags,
-                                            location=location)
-    return client.graph_query.create_or_update(resource_group_name=resource_group_name,
-                                               resource_name=resource_name,
-                                               properties=graph_shared_query)
+    result_dict = dict()
+    result_dict['data'] = results
+    result_dict['skip_token'] = skip_token
+    return result_dict
```

Comment:
Why not directly return response?

Reply:
Actually that would work. Changed to return response.

Reply:
We don't want to expose extra info to the user like facets, which is available in API but there is no support for it in CLI. This way is kind of trimming the response to the values that are within CLI scope.